wikiclass/feature_lists/wikidatawiki.py

```diff
@@ -21,6 +39,137 @@ class items:
     """
     HUMAN = 'Q5'
 
+def _process_all_sources(item):
+
+	list_sources_in_JSON = []
+	
+	try:
+		for property in item.claims:
+			list_of_properties = property
+			for claim in item.claims[list_of_properties]:
+				list_of_claims = claim
+				for i, source in enumerate(list_of_claims.sources):
+					for index_list_1, index_list_2 in source.items(): 
+						sources_in_JSON = index_list_2[0].toJSON()
+						list_sources_in_JSON.append(sources_in_JSON['datavalue']['value'])
+		
+		return list_sources_in_JSON
+	except:
+		return 0 #if the revision does not contain any sources, return 0
+
+def _process_wikimedia_sources(item):
+
+	list_sources_in_JSON = []
+	list_wikimedia_sources = []
+	
+	for property in item.claims:
+		list_of_properties = property
+		for claim in item.claims[list_of_properties]:
+			list_of_claims = claim
+			for i, source in enumerate(list_of_claims.sources):
+				for index_list_1, index_list_2 in source.items(): 
+					sources_in_JSON = index_list_2[0].toJSON()
+					list_sources_in_JSON.append(sources_in_JSON['datavalue']['value'])
+    
+	wdir_path = os.path.dirname(os.path.realpath(__file__)) #get the current working directory path
+	csv_path = os.path.join(wdir_path, 'excluded_qids.csv')
+
+	for list_sources_in_JSON_content in list_sources_in_JSON: 
+		with open(csv_path) as csvfile:
```

Comment:
Re. "Also, read the file once and store the items in a set". 
I think it is better to store duplicate in this case. Suppose there are 2 revisions: 
- Revision A with 1 English Wikipedia source.
- Revision B with 10 English Wikipedia source.

I think that Revision B is better than Revision A because Revision B has more sources than Revision A even though all of Revision B's sources are coming from the same English Wikipedia source